src/test/java/org/openelisglobal/gender/GenderServiceTest.java

```diff
@@ -0,0 +1,80 @@
+package org.openelisglobal.gender;
+
+import java.sql.Timestamp;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openelisglobal.BaseWebContextSensitiveTest;
+import org.openelisglobal.gender.service.GenderService;
+import org.openelisglobal.gender.valueholder.Gender;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class GenderServiceTest extends BaseWebContextSensitiveTest {
+
+    @Autowired
+    GenderService genderService;
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Before
+    public void init() throws Exception {
+        executeDataSetWithStateManagement("testdata/gender.xml");
+    }
+
+    @Test
+    public void verifyTestData() {
+        List<Gender> genders = genderService.getAll();
+        System.out.println("Genders we have in db: " + genders.size());
+        genders.forEach(gender -> System.out
+                .println(gender.getId() + " - " + gender.getGenderType() + " - " + gender.getDescription()));
+
+    }
+
+    @Test
+    public void createGender_shouldCreateNewGender() throws Exception {
+        List<Gender> genders = genderService.getAll();
+        boolean genderExists = genders.stream().anyMatch(g -> "M".equals(g.getGenderType()));
+
+        if (!genderExists) {
```

Comment:
so if the gender exists then we will never test the `createGender` functionality?